### PR TITLE
fix PR 30843

### PR DESCRIPTION
**Pull request by jianzhangbjz**
Address errors in PR #30843

### CODE REVIEW

File: ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh [hunk 1]
```diff
@@ -198,7 +198,7 @@ spec:
 EOF
     set +e 
     COUNTER=0
-    while [ $COUNTER -lt 30 ]
```
Comment (jianlinliu) [inline ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh:201]: You want the `COUNTER` stand for the total timeout or the retry times?

**Comment by jianlinliu** [comment]
I strongly fix https://github.com/openshift/release/pull/30843/files#r930617318 together

File: ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh (hunk 1 above)
Comment (jianzhangbjz) [reply ci-operator/step-registry/enable-qe-catalogsource/disconnected/enable-qe-catalogsource-disconnected-commands.sh:201]: Yes,

**Comment by jianzhangbjz** [comment]
> I strongly fix https://github.com/openshift/release/pull/30843/files#r930617318 together

Done.

**Comment by jianlinliu** [comment]
From https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/30845/rehearse-30845-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-e2e-azure-upi-proxy-p2/1552179400865222656, seem like other place also need some similar enhancement when doing some global update.

**Comment by jianzhangbjz** [comment]
Hi @jianlinliu for the below error, it really related to the cluster performance of Prow, that's why the developers have to retest their feature PRs again and again. If we add trying steps for those commands, that means all `oc` client related commands should be enhanced.  I'd suggest that we skip it. 
```yaml
error: error when creating "STDIN": Post "https://api.ci-op-mkcscgdb-b0a5d.qe.azure.devcluster.openshift.com:6443/apis/operator.openshift.io/v1alpha1/imagecontentsourcepolicies?fieldManager=kubectl-create&fieldValidation=Ignore": dial tcp 20.236.77.92:6443: i/o timeout
```

**Comment by jianlinliu** [comment]
From the log, the error was thrown from cluster api, I think we still need to enhance it if such "flake" are reproduced frequently in the future.  To be more safe, maybe we can add "sleep 5" between `update_global_auth` and `create_catalog_sources` to make it    a bit more robust.

**Comment by jianzhangbjz** [comment]
> I think we still need to enhance it if such "flake" are reproduced frequently in the future. To be more safe, maybe we can add "sleep 5" between update_global_auth and create_catalog_sources to make it a bit more robust.

But, adding `sleep` won't solve this issue. If we really want to solve it we should improve the cluster performance.

**Comment by jianzhangbjz** [comment]
Anyway, I added it. But, it's really ugly. We should NOT address this kind of by adding the `sleep` always. I should have not submitted that PR #30843

**Comment by jianlinliu** [comment]
> We should NOT address this kind of issue by adding the sleep always. I should have not submitted that PR https://github.com/openshift/release/pull/30843

Sorry, I can not agree with that. We already noticed some issues from code quality perspective, we should try our best to improve it a bit to make it better. If we leave such "flake" (actually I even doubt it is really a flake) there untouched, will block the whole testing, if such "flake" is seen more and more, QE would never complete our test plan. Our final goal is running all testing in prow ci, we have no other choice, unless we assert prow ci is not a good place to run QE's ci testing.

**Comment by jianzhangbjz** [comment]
And, the reason why I don't think it's a good solution is that even if this step passed after adding the `trying` or `sleeping` workaround. But, there is already something wrong with the cluster. This cluster is not stable anymore, and won't handle the request sometimes. That means the following e2e test case will fail due to the cluster issue. It's very noisy for each team to debug it. I don't think each team will add the `sleep` step between two `oc` commands for their e2e test case.

**Comment by jianlinliu** [comment]
> If we really want to solve it we should improve the cluster performance.

This is a vague conclusion. Is there any specific suggestion to improve the cluster's "Performance" to move forwards our e2e testing?

**Comment by jianlinliu** [comment]
> This cluster is not stable anymore, and won't handle the request sometimes. That means the following e2e test case will fail due to the cluster issue. It's very noisy for each team to debug it. I don't think each team will add the sleep step between two oc commands for their e2e test case.

Just like what we discuss in slack, intermittently api unavailable is very common for OCP cluster, especially when updating cluster's global configuration (including machineconfig, pullsecret), we resolved many similar issues in flexy codes via retry/wait workaround, though sometimes it is not most ideal, but that is how we survive till now.

**Comment by jianlinliu** [comment]
> > If we really want to solve it we should improve the cluster performance.
> 
> This is a vague conclusion. Is there any specific suggestion to improve the cluster's "Performance" to move forwards our e2e testing?

I also checked the master vm size, whatever in prow ci and jenkins ci, both of them are using "Standard_D8s_v3" as the default vm size, if we did not see such issues in jenkins ci, then we also should not see such issue in prow ci, if you doubt the "performance" issue is because of vm size.

**Comment by jianlinliu** [comment]
And one more thing need your attention,  in prow ci script, "set -e" is enabled in most of cases, while in flexy, we did not force that by default. That means prow script has higher requirement to code quality, while does that mean flexy code never hit such issues before? I do not think so, just many scripts ignore the api unavailable issues in flexy. During our daily ci watcher work, we saw api unavailable issues for several times in jenkins log, especially in the called scripts in flexy's post-actions. To make flexy more stable, we put `wait_for_stable_cluster_status` (wait/retry is necessary in the function) in several places in post-actions to ensure the installed cluster get into stable status before deliver it to users.

**Pull request merged**